search/api.py

```diff
@@ -84,11 +91,18 @@ def execute_search(*, user, query):
     Returns:
         dict: The Elasticsearch response dict
     """
-    from channels.api import Api
-
     index = get_default_alias_name(ALIAS_ALL_INDICES)
     search = Search(index=index, using=get_conn())
     search.update_from_dict(query)
-    channel_names = [channel.display_name for channel in Api(user).list_channels()]
-    search = search.filter("terms", channel_name=channel_names)
+    channel_names = list(
+        ChannelGroupRole.objects.filter(
+            group__user=user, role__in=(ROLE_CONTRIBUTORS, ROLE_MODERATORS)
+        )
+        .values_list("channel__name", flat=True)
+        .distinct()
+    )
```

Comment:
This will prevent subscribers of restricted channels from getting search results (checkbox 2 in #1461).  Using `list(profiles.api.get_channels(user))` will include the channel names that the user subscribes to.

Reply:
I think the only way we can support that is to add the `channel_type` information to the `Channel` model and then update the filter to permit all searches to those channels. I'm going to do that in the next PR and then update this logic to fix the filtering

Reply:
OK, I tried `list(profiles.api.get_channels(user))` and that does work for searching within a restricted channel when logged in as a subscribed user, but I'm fine with solving it another way in a future PR.